src/compiler/checker.ts

```diff
@@ -24551,16 +24551,19 @@ namespace ts {
         function getJsxManagedAttributesFromLocatedAttributes(context: JsxOpeningLikeElement, ns: Symbol, attributesType: Type) {
             const managedSym = getJsxLibraryManagedAttributes(ns);
             if (managedSym) {
-                const declaredManagedType = getDeclaredTypeOfSymbol(managedSym);
+                const declaredManagedType = getDeclaredTypeOfSymbol(managedSym); // fetches interface type, or initializes symbol links type parmaeters
                 const ctorType = getStaticTypeOfReferencedJsxConstructor(context);
+                if (managedSym.flags & SymbolFlags.TypeAlias) {
+                    const params = getSymbolLinks(managedSym).typeParameters;
+                    if (length(params) >= 2) {
+                        const args = fillMissingTypeArguments([ctorType, attributesType], params, 2, isInJSFile(context));
+                        return getTypeAliasInstantiation(managedSym, args);
+                    }
+                }
                 if (length((declaredManagedType as GenericType).typeParameters) >= 2) {

```

Comment:
Hi, I'm the reporter of this issue - thanks for the swift fix! I was really curious why this has been happening (because seemingly it didn't really look that much different from other code that has been working) and I got into reading this PR to see how it got fixed.

I'm obviously just a bystander and I lack *a lot* of context but understanding this fix probably would allow me to understand the TS compiler better. I'm wondering if you would be able to share a little bit more in-depth explanation behind this? I will totally understand if you won't have time for that though.

I suspect that `SymbolFlags.TypeAlias` checks if this is a `type` type and not the `interface` - does it mean that one could use the `interface` for `LibraryManagedAttributes`? I kinda assumed that this is not possible but maybe I was wrong - I can't quite check this quickly myself because this has not been yet released to the nightly channel.

This change later checks the `typeParameters.length >= 2` - this looks quite odd for me because the given code uses exactly 2 type parameters, so why this is even relevant for this fix? I would more expect that this piece of code would provide exactly as many type parameters as the `managedSym` accepts (up to 2 as no more will be ever provided). Shouldn't also maybe `JSX.LibraryManagedAttributes` with more than 2 type parameters result in an error since those extras won't ever be provided? And how "fillMissingTypeArguments" is relevant here for the "top most" `managedSym`? Will this fix the other case that I've mentioned (albeit without repro because I've concluded that the underlying issue is probably the same):
```ts
type LibraryManagedAttributes<C, P> = WithConditionalCSSProp<
  ReactJSXLibraryManagedAttributes<C, P>
>
```
This snippet is using both type parameters so I would expect that nothing is missing here but yet it doesn't work either.

I was wondering about an additional thing for quite a bit - maybe you could describe this shortly as well. What's the difference between `aliasTypeArguments` and `typeArguments`, between `symbol` and `aliasSymbol`, between `type` and `typeAlias`? 🤔

Reply:
The `aliasTypeArguments` list is the type arguments plugged in to the type alias symbol associated with the type - which, in many cases, is the `LibraryManagedAttributes` symbol, _however_, if the target of the type alias already has an alias symbol itself (eg, `WithConditionalCSSProp` is a type alias), then it'll be the arguments for _that_, since the type is first made via that alias.